### ISSUE TITLE
BoardCollection.Add() returns 'invalid value for name'

### ISSUE THREAD

**Issue by dwaynew** [gregsdennis/Manatee.Trello#140]
Hi, thanks for a great library! I'm trying to create a Board using the following code:
```
var organisation = Member.Me.Organizations.First();
var board = organisation.Boards.Add("TEST);
```
what I get returned is the following error message:
```
Trello has reported an error with the request: invalid value for name
```

I suspect this is because the Trello API docs specify the `name` should be part of the querystring rather than the body (https://trello.readme.io/reference#boardsid):
```
curl --request POST --url 'https://api.trello.com/1/boards/?name=name'
```

This looks like a common pattern for creation of for eg Lists, Cards. Any chance there is a workaround or fix floating around? I'm on 2.0.0 of the libs (the later ones gave me compile errors due to the nuget versioning).

**Comment by gregsdennis** [comment]
I have no problems creating a new board (and subsequently deleting it).  Would you mind emailing me the request?  You can capture it using Fiddler or similar.  My email is my username here at yahoo.

Thanks for reporting the issue.

**Comment by dwaynew** [comment]
Hi, this is my request (identical to yours):
```
POST https://trello.com/1/boards?key=<KEY>&token=<TOKEN> HTTP/1.1
Content-Type: text/plain; charset=utf-8
Host: trello.com
Content-Length: 59

{"name":"TEST","idOrganization":"<ID>"}
```
and the response:
```
HTTP/1.1 400 Bad Request
X-DNS-Prefetch-Control: off
X-Frame-Options: DENY
Strict-Transport-Security: max-age=15552000
X-Download-Options: noopen
Surrogate-Control: no-store
Cache-Control: max-age=0, must-revalidate, no-cache, no-store
Pragma: no-cache
Expires: Thu, 01 Jan 1970 00:00:00
X-Content-Type-Options: nosniff
Referrer-Policy: strict-origin-when-cross-origin
X-XSS-Protection: 1; mode=block
X-Trello-Version: 5
X-Trello-Environment: Production
Access-Control-Allow-Origin: *
Access-Control-Allow-Methods: GET, PUT, POST, DELETE
Access-Control-Allow-Headers: Authorization, Accept, Content-Type
Content-Type: text/plain; charset=utf-8
Content-Length: 22
ETag: W/"16-cac2b940"
Vary: Accept-Encoding
Date: Mon, 05 Mar 2018 09:48:50 GMT
Connection: close

invalid value for name
```
Any ideas?

**Comment by gregsdennis** [comment]
@dwaynew Um.... I got your post via email, and I got nothing.  I think I'm going to have to send this to Trello to get their input.  They're probably going to ask for the IDs and tokens in the request.

**Comment by dwaynew** [comment]
@gregsdennis after capturing the request I noticed what I think was the issue. The Content-Type was being set to text/plain:
```
POST https://trello.com/1/boards?key=<KEY>&token=<TOKEN> HTTP/1.1
Content-Type: text/plain; charset=utf-8
Host: trello.com
Content-Length: 59

{"name":"TEST","idOrganization":"<ID>"}
```
This was with v2.0.0 of Manatee.Trello.WebApi from nuget. I tried the same operation just now but with a reference to the latest code from the master branch and it sets the header correctly:
```
POST https://trello.com/1/boards?key=<KEY>&token=<TOKEN> HTTP/1.1
Content-Type: application/json
Host: trello.com
Content-Length: 59
Expect: 100-continue
Connection: Keep-Alive

{"name":"TEST","idOrganization":"<ID>"}
```
and the Board gets created no problem! So I guess something in that older version? Incidentally I was just on the older version due to getting some version conflict trying to build on the latest from nuget?

Thanks for the pointer in the right direction!

**Comment by gregsdennis** [comment]
Yeah, I just noticed that as well.  It looks like that was a fix in the 2.0.1 version of the WebApi package.

Glad you got it up and running.
